mlflow/utils/autologging_utils.py

```diff
@@ -533,6 +543,213 @@ def end_session(cls):
         cls._session = None
 
 
+class AutologgingEventLogger:
```

Comment:
@apurva-koti @harupy This abstraction allows developers (e.g., Databricks engineers) to plug in their own event loggers in order to instrument autologging ML execution